pkg/scheduler/scheduler.go

```diff
@@ -285,36 +286,49 @@ func findFlavorForResource(
 	wUsed map[string]int64,
 	resourceFlavors map[string]*kueue.ResourceFlavor,
 	cq *cache.ClusterQueue,
-	spec *corev1.PodSpec) (string, int64) {
+	spec *corev1.PodSpec) (string, int64, []error) {
+	var errs []error
+
+	if _, exists := cq.RequestableResources[name]; !exists {
+		errs = append(errs, fmt.Errorf("resource %s not found", name))
+		return "", 0, errs
+	}
+
 	// We will only check against the flavors' labels for the resource.
 	selector := flavorSelector(spec, cq.LabelKeys[name])
 	for _, flvLimit := range cq.RequestableResources[name] {
 		flavor, exist := resourceFlavors[flvLimit.Name]
 		if !exist {
-			log.Error(nil, "Flavor not found", "Flavor", flvLimit.Name)
+			// TODO: if "flavor not found" error emitted too frequently, we'd like to aggregate them into one error.
+			errs = append(errs, fmt.Errorf("flavor %s not found", flvLimit.Name))
 			continue
 		}
-		_, untolerated := corev1helpers.FindMatchingUntoleratedTaint(flavor.Taints, spec.Tolerations, func(t *corev1.Taint) bool {
+		taint, untolerated := corev1helpers.FindMatchingUntoleratedTaint(flavor.Taints, spec.Tolerations, func(t *corev1.Taint) bool {
 			return t.Effect == corev1.TaintEffectNoSchedule || t.Effect == corev1.TaintEffectNoExecute
 		})
 		if untolerated {
+			errs = append(errs, fmt.Errorf("flavor %s holds untolerated taint %s", flvLimit.Name, taint))
 			continue
 		}
 		if match, err := selector.Match(&corev1.Node{ObjectMeta: metav1.ObjectMeta{Labels: flavor.Labels}}); !match || err != nil {
 			if err != nil {
-				log.Error(err, "Matching workload affinity against flavor; no flavor assigned")
-				return "", 0
+				errs = append(errs, fmt.Errorf("flavor %s match node affinity error: %s", flvLimit.Name, err))
```

Comment:
this would be an actual error. Maybe we should return it separately. Or the status can have an error field, separate from reasons.